src/sedpack/io/dataset_iteration.py

```diff
@@ -93,6 +100,19 @@ def shard_paths_dataset(
         if shards:
             shards_list = shards_list[:shards]
 
+        # Only use a limited amount of shards for each setting of
+        # custom_metadata.
+        if custom_metadata_type_limit:
+            counts: dict[tuple[tuple[str, Any], ...], int] = {}
+            old_shards_list = shards_list
+            shards_list = []
+            for shard_info in old_shards_list:
+                k = tuple(sorted(shard_info.custom_metadata.items()))
+                counts[k] = counts.get(k, 0) + 1
+                if counts[k] <= custom_metadata_type_limit:
+                    shards_list.append(shard_info)
+            self._logger.info("Took %s shards total", len(shards_list))
```

Comment:
why not using a f-string here?

Reply:
Performance or keeping pylint from complaining :)

https://pylint.readthedocs.io/en/stable/user_guide/messages/warning/logging-fstring-interpolation.html

Reply:
ack